toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,94 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(phi: np.ndarray | list[np.ndarray | list[np.ndarray]], tol: float = 1e-9) -> bool:
+    r"""Determine whether a quantum channel is extremal.
+
+    (Section 2.2.4: Extremal Channels from :cite:`Watrous_2018_TQI`).
+
+    Theorem 2.31 in :cite:`Watrous_2018_TQI` provides the characterization of extremal
+    quantum channels: A channel :math:`\Phi` is an extreme point of the convex set
```

Comment:
```suggestion
    quantum channels as a channel :math:`\Phi` is an extreme point of the convex set
```

making a minor change because something is wrong with the workflow